OpenCL_C.txt

```diff
@@ -2335,47 +2229,394 @@ defined as the value of the <<opencl-device-queries,
 `CL_DEVICE_MAX_CONSTANT_ARGS` device query>>.
 ====
 
-Variables in the program scope can be declared in the `+__constant+` address
-space.
-Variables in the outermost scope of kernel functions can be declared in the
-`+__constant+` address space.
-These variables are required to be initialized and the values used to
-initialize these variables must be a compile time constant.
 Writing to such a variable results in a compile-time error.
 
+Example:
+
+[source,c]
+----------
+constant int a = 3; // int in constant address space initialized with a constant value.
+kernel void k1(global int *buf)
+{
+  buf[a] = ...; // allowed. All work items access element with index 3;
+}
+kernel void k2(global int *buf)
+{
+  *buf = a; // allowed. All work items stored value 3;
+  a = 42; // error. a is in constant memory.
+}
+----------
+
+
 Implementations are not required to aggregate these declarations into the
-fewest number of constant arguments.
-This behavior is implementation defined.
+fewest number of constant arguments. This behavior is implementation defined.
 
 Thus portable code must conservatively assume that each variable declared
 inside a function or in program scope allocated in the `+__constant+`
 address space counts as a separate constant argument.
---
 
+--
 
 [[private-or-private]]
 === `+__private+` (or `private`)
 
 [open,refpage='private',desc='private Address Space Qualifiers',type='freeform',spec='clang',anchor='private-or-private',xrefs='addressSpaceQualifiers constant genericAddressSpace global local']
 --
 
-Variables inside a kernel function not declared with an address space
-qualifier, all variables inside non-kernel functions, and all function
-arguments are in the `+__private+` or `private` address space.
---
+Private address space is a memory segment that can only be accessed by one work
+item. Variables that are not shareable among work items are allocated in private
+and it is the default address space for most of variables in particular variables
+with automatic storage duration.
```

Comment:
Would it be reasonable to refer to the Inference section (i.e. addr-spaces-inference) where it is described for all variables and address spaces in details? I would prefer to avoid repeating the context in order to prevent possible inconsistency in future modifications.